nameserver/server.go

```diff
@@ -194,17 +206,6 @@ func (s *DNSServer) Start() error {
 		Info.Printf("[dns] Cache: %d entries", s.cache.Capacity())
 	}
 
-	pc, err := net.ListenPacket("udp", s.ListenAddr)
-	if err != nil {
-		return err
-	}
-	s.pc = pc
-
-	_, port, err := net.SplitHostPort(pc.LocalAddr().String())
-	if err != nil {
-		return err
-	}
-	s.ListenAddr = fmt.Sprintf(":%s", port)
```

Comment:
Yes, we call `Start()` on a different goroutine, because it blocks.  Hence a race condition is reported, although it is not likely to do harm.

Not clear that creating a listener can be called a side-effect. It's an essential part of the server.

Maybe separating out a third `ListenAndServe()` routine which blocked would make it clearer?